inc/azure_c_util/interlocked_if.h

```diff
@@ -0,0 +1,156 @@
+#ifndef INTERLOCKEDIFIF_H
+#define INTERLOCKEDIFIF_H
+
+#include <stdint.h>
+
+long InterlockedIFAdd( 
```

Comment:
>long [](start = 0, length = 4)

if stdint is included, then I suppose all the types should be ones from stdint.h

Also comes with the stability of "sizeof" for types (int might be 4 or 8... "depending"). types from stdint are fixed sizeof.

For example, long here should read "int32_t"

Reply:
I've changed all occurrences of `long` to `int32_t`, `short` to `int16_t` and `char` to `int8_t` as suggested by Matt. Does that work?